services/user-service/api/handler/login_test.go

```diff
@@ -0,0 +1,178 @@
+package handler
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/hingew/hsfl-master-ai-cloud-engineering/lib/gomockhelpers"
+	mocks "github.com/hingew/hsfl-master-ai-cloud-engineering/user-service/_mocks"
+	"github.com/hingew/hsfl-master-ai-cloud-engineering/user-service/user/model"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+)
+
+func TestLoginHandler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	userRepository := mocks.NewMockRepository(ctrl)
+	hasher := mocks.NewMockHasher(ctrl)
+	tokenGenerator := mocks.NewMockTokenGenerator(ctrl)
+	handler := NewLoginHandler(userRepository, hasher, tokenGenerator)
+
+	t.Run("should return 405 METHOD NOT ALLOWED if method is not POST", func(t *testing.T) {
+		// given
+		w := httptest.NewRecorder()
+		r := httptest.NewRequest("GET", "/api/v1/auth/login", nil)
+
+		// when
+		handler.ServeHTTP(w, r)
+
+		// then
+		assert.Equal(t, http.StatusMethodNotAllowed, w.Code)
+	})
```

Comment:
Why is the status code 405 and not 404?

Reply:
da die HTTP-Methode POST hier verwendet wird, andere HTTP-Methoden sollten nicht erlaubt sein

Reply:
Ich verstehe schon das hier ein POST verwendet wird, aber nicht warum nicht ein 404 zurückkommt.

Reply:
Weil `StatusMethodNotAllowed` ein 405 ist, kein `404 page not found`. Angenommen wir stellen über Postman eine OPTION-Anfrage, dann sollte ein 405 erscheinen.

Reply:
We will address this in https://github.com/hingew/hsfl-master-ai-cloud-engineering/issues/14.